astropy/convolution/convolve.py

```diff
@@ -159,16 +159,19 @@ def convolve(array, kernel, boundary='fill', fill_value=0.,
 
     # Check that the arguments are lists or Numpy arrays
 
-    if isinstance(array, list):
-        array_internal = np.array(array, dtype=float)
-        array_dtype = array_internal.dtype
-    elif isinstance(array, np.ndarray):
+    if isinstance(array, np.ndarray):
         # Note this won't copy if it doesn't have to -- which is okay
         # because none of what follows modifies array_internal.
         array_dtype = array.dtype
         array_internal = array.astype(float, copy=False)
     else:
-        raise TypeError("array should be a list or a Numpy array")
+        try:
+            array_internal = np.array(array, dtype=float)
+        except Exception as e:
```

Comment:
Can we catch a more specific type here?  I don't like general exception catching because it can sometimes catch things it wasn't meant to.

Reply:
Maybe ``TypeError`` and ``ValueError``?

```
In [3]: np.array('banana', dtype=float)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-a46e4cf2c8fe> in <module>()
----> 1 np.array('banana', dtype=float)

ValueError: could not convert string to float: 'banana'

In [4]: np.array(set())
Out[4]: array(set(), dtype=object)

In [5]: np.array(set(), dtype=float)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-ed9cb48f6d39> in <module>()
----> 1 np.array(set(), dtype=float)

TypeError: float() argument must be a string or a number, not 'set'
```

Reply:
I see your point here - unclear if `ValueError` or `TypeError` are a *complete* set for this case, but I guess those are the sets that map best onto "the input type is wrong".  Any other exception just passes through anyway, I suppose.